Plotting/composite_plots.py

```diff
@@ -0,0 +1,211 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from astropy.io import ascii as asc
+from matplotlib.backends.backend_pdf import PdfPages
+from Metallicity_Stack_Commons import OIII_r
+from Metallicity_Stack_Commons.column_names import temp_metal_names0, bin_ratios0, bin_mzevolve_names0
+from Metallicity_Stack_Commons.column_names import bin_names0, filename_dict
+from .relation_fitting import extract_error_bars
+
+
+
+def bin_derived_props_plots(fitspath, hbeta_bin = False, err_bars = False, revised = False):
+    '''
+    Purpose: 
+        This function creates plots for the stacked spectra (bins): OII/HBeta vs Stellar Mass,
+        OIII/HBeta vs Stellar Mass, OIII4363/OIII5007 vs Stellar Mass, R23 vs Temperature, O32 vs Temperature,
+        Metallicity vs R23, Metallicity vs O32, Temperature vs Stellar Mass, and Metallicity vs Stellar Mass.
+        
+    Params:
+        fitspath --> string of the file path where data files are located. 
+        hbeta_bin (OPTIONAL) --> if the binning type is mass-LHbeta bins, then hbeta_bin = True.
+        err_bars (OPTIONAL) --> if error bars for metallicity and temperature should be plotted, then 
+                                err_bars = True.
+        revised (OPTIONAL) --> if the revised data tables should be used, then revised = True.
+        
+    Returns:
+        None
+        
+    Outputs:
+        pdf_pages --> a pdf containing all of the plots for the stacked spectra on separate pages.
+    '''
+    
+    #Read in data tables
+    if revised:
+        pdf_pages = PdfPages(fitspath + filename_dict['bin_derived_prop_rev'].replace('.tbl', '.pdf'))
+        metal_table = asc.read(fitspath + filename_dict['bin_derived_prop_rev'])
+        em_table = asc.read(fitspath + filename_dict['bin_fit_rev'])
+        valid_table = asc.read(fitspath + filename_dict['bin_valid_rev'])
+    else:
+        pdf_pages = PdfPages(fitspath + filename_dict['bin_derived_prop'].replace('.tbl', '.pdf'))
+        metal_table = asc.read(fitspath + filename_dict['bin_derived_prop'])
+        em_table = asc.read(fitspath + filename_dict['bin_fit'])
+        valid_table = asc.read(fitspath + filename_dict['bin_valid'])
+    bin_table = asc.read(fitspath + filename_dict['bin_info'])
+    
+    #Read in composite values
+    com_O_log = metal_table[temp_metal_names0[1]].data
+    T_e = metal_table[temp_metal_names0[0]].data
+    logR23 = metal_table[bin_ratios0[0]].data
+    logO32 = metal_table[bin_ratios0[1]].data
+    log_twoBeta = np.log10(metal_table[bin_ratios0[2]].data)
+    log_threeBeta = np.log10(metal_table[bin_ratios0[3]].data)
+    OIII4363 = em_table['OIII_4363_Flux_Observed'].data
+    OIII5007 = em_table['OIII_5007_Flux_Observed'].data
+    logM_avg = bin_table[bin_mzevolve_names0[2]].data
+    logLHb_avg = bin_table[bin_mzevolve_names0[6]].data
+    detection = valid_table[bin_names0[2]].data
+    
+    #OIII ratio                
+    log_OIIIratio = np.log10(OIII4363 / (OIII5007 * (1 + 1/OIII_r)))
```

Comment:
I think this is better derived using the R_temp_calc or the `flux_ratios_dict` from MSC's `ratios.flux_ratios()`